src/app/course-list/course/component.ts

```diff
@@ -26,4 +30,12 @@ export class CourseComponent {
     }
     return outstr;
   }
+
+  public isSelected(section : Section) : boolean {
+    return this.selectService.isSectionSelected(section);
+  }
+
+  public clickSection(section : Section) {
+    this.isSelected(section) ? this.selectService.removeSection(section) : this.selectService.addSection(section);
```

Comment:
Maybe make a function for toggling selection, so you dont have to make 3 calls to the service